specifyweb/frontend/js_src/lib/components/SchemaConfig/Tables.tsx

```diff
@@ -75,18 +75,33 @@ export function TableList({
 }: {
   readonly cacheKey: CacheKey;
   readonly getAction: (table: SpecifyTable) => string | (() => void);
-  readonly filter?: (table: SpecifyTable) => boolean;
+  readonly filter?: (showHiddenTables: boolean, table: SpecifyTable) => boolean;
   readonly children?: (table: SpecifyTable) => React.ReactNode;
 }): JSX.Element {
   const [showHiddenTables = false, setShowHiddenTables] = useCachedState(
     cacheKey,
     'showHiddenTables'
   );
 
+  const advancedTables = React.useMemo(() => {
+    const allTables = Object.values(genericTables);
+
+    const filterFunction: ((table: SpecifyTable) => boolean) | undefined =
+      filter?.bind(undefined, showHiddenTables) ??
+      (showHiddenTables ? undefined : ({ isSystem }): boolean => !isSystem);
+
+    const filteredTables =
+      typeof filterFunction === 'function'
+        ? allTables.filter(filterFunction)
+        : allTables;
+
+    return filteredTables.sort(sortFunction(({ name }) => name));
+  }, [filter, showHiddenTables]);
+
```

Comment:
why do all of this here?
couldn't this be much simpler - in `sortedTables`, instead of always passing false to `filter` function, just pass showHiddenTables